enhancements/installer/openstack-ipi-failure-domains.md

```diff
@@ -0,0 +1,254 @@
+---
+title: openstack-ipi-failure-domains
+authors:
+  - "@EmilienM"
+  - "@mandre"
+  - "@mdbooth"
+reviewers:
+  - "@sdodson"
+  - "@patrickdillon"
+approvers:
+  - "@sdodson"
+  - "@patrickdillon"
+api-approvers:
+  - "None"
+creation-date: 2022-07-04
+last-updated: 2022-07-05
+tracking-link:
+  - "https://issues.redhat.com/browse/OSASINFRA-2908"
+see-also:
+  - "/enhancements/installer/aws-customer-provided-subnets.md"
+  - https://github.com/openshift/enhancements/pull/918
+superseded-by:
+  - "https://docs.google.com/document/d/1_AjhGEyK6vGBFSSqiLVtLt_qOq6o9t2dtKqq2eFr48Y/"
+---
+
+# Allow to stretch a cluster across multiple failure domains
+
+## Summary
+
+For organizations where service availability is a concern, networking architectures are usually more modern than what we can find in a traditional datacenter.
+For example, the "Spine and Leaf" design is a recommended choice when deploying OpenStack at scale.
+In this architecture, each leaf (lower-level access switch) is connected to multiple spines (upper-level core switch) in a full mesh. The latency becomes predictable, there is no more bottleneck and overall the resilience is improved.
+Each leaf brings their own network fabrics that are usually using their own power units. Also each network fabric manages their its own Layer-2 domain, and therefore network subnets. Routing protocols are usually used so applications can communicate each other in the datacenter, since they don't live within the same domain anymore.
```

Comment:
Do we have any performance concerns with either control or data plane with this new architecture?  Will we verify that performance between this new architecture and the original are comparable?

Reply:
We'll probably have to define a maximum latency and a minimum bandwidth that we know the cluster will stay healthy.

Reply:
I'm adding it to the `Open Questions` section.